src/test/scala/org/apache/spark/deploy/armada/e2e/TestOrchestrator.scala

```diff
@@ -93,40 +94,78 @@ class TestOrchestrator(
       armadaClient
     )
 
-    // Wait for Spark driver pod to be scheduled and running
-    val driverSelector  = s"spark-role=driver,test-id=${context.testId}"
-    var driverRunning   = false
-    var attempts        = 0
-    val maxWaitAttempts = 60 // 2 minutes max to wait for driver
+    if (!isClientMode) {
+      // Wait for Spark driver pod to be scheduled and running
+      val driverSelector  = s"spark-role=driver,test-id=${context.testId}"
+      var driverRunning   = false
+      var attempts        = 0
+      val maxWaitAttempts = 60 // 2 minutes max to wait for driver
 
-    while (!driverRunning && !jobCompleted() && attempts < maxWaitAttempts) {
-      try {
-        val driverPods = Await.result(
-          k8sClient.getPodsByLabel(driverSelector, context.namespace),
-          10.seconds
-        )
-        driverPods.headOption match {
-          case Some(pod) if pod.getStatus != null && pod.getStatus.getPhase == "Running" =>
-            driverRunning = true
-          case Some(_) =>
-          case None    =>
+      while (!driverRunning && !jobCompleted() && attempts < maxWaitAttempts) {
+        try {
+          val driverPods = Await.result(
+            k8sClient.getPodsByLabel(driverSelector, context.namespace),
+            10.seconds
+          )
+          driverPods.headOption match {
+            case Some(pod) if pod.getStatus != null && pod.getStatus.getPhase == "Running" =>
+              driverRunning = true
+            case Some(_) =>
```

Comment:
Are the cases for Some and None matches needed at all, if we aren't doing anything here?